python/social_bot/teacher_tasks.py

```diff
@@ -205,3 +208,462 @@ def set_goal_name(self, goal_name):
         logging.debug('Setting Goal to %s', goal_name)
         self._goal_name = goal_name
 
+
+@gin.configurable
+class GroceryGroundGoalTask(GoalTask):
```

Comment:
Not sure if it is a good choice to directly move all the tasks from grocery_ground.py to teacher_tasks.py. As after doing this, some of the tasks in teacher_tasks.py are highly coupled with some specific world/env.
The ideal case is to only keep relatively general tasks that can be reused in different worlds or environment in teacher_tasks.py while leaving those coupled with specific env into separate files.

Reply:
Yes, coupled with specific env tasks should leave to the env file. I was considering to decouple the 3 tasks with grocery env. Goal task and kicking-ball task could be more general. ICubAuxTask is also supposed to be a general helper task providing reward-shaping for walking by 2 legs.
I was also considering rename GroceryGround to PlayGround and remove it's dependency on specific task. So that we can assemble different tasks and different morphology of teachers in the playground through gin. Too many changes in one pull request are hard to review. Let's open another PR for these.

Reply:
Sounds good. We can do this progressively.